### PR TITLE
Fix GitHub callback URL documentation

### LINKED ISSUE #14360: Wrong callback URL in github auth provider docs
The github auth provider documentation at https://backstage.io/docs/auth/github/provider#create-an-oauth-app-on-github specifies `http://localhost:7007/api/auth/github` as the callback url to use when creating the github oauth app, which has the incorrect path. The correct value (documented at https://backstage.io/docs/getting-started/configuration#setting-up-authentication in the getting started guide) is `http://localhost:7007/api/auth/github/handler/frame`.

Using the incorrect path will result in one of two problems:
 * If the provider is configured with a matching `callbackUrl`, backstage will return a 404 with a provider not found message.
 * If the provider uses the default callback url, github will return an error about a callback mismatch.

(Since this is a pretty clear documentation issue, I don't believe the usual bug report template is relevant.)

### PR DESCRIPTION
**Pull request by jgrumboe**
## Hey, I just made a Pull Request!

This fixes #14360 

#### :heavy_check_mark: Checklist

- [x] A changeset describing the change and affected packages. ([more info](https://github.com/backstage/backstage/blob/master/CONTRIBUTING.md#creating-changesets))
- [x] Added or updated documentation
- [ ] Tests for new functionality and regression tests for bug fixes
- [ ] Screenshots attached (for UI changes)
- [x] All your commits have a `Signed-off-by` line in the message. ([more info](https://github.com/backstage/backstage/blob/master/CONTRIBUTING.md#developer-certificate-of-origin))

### CODE REVIEW

**Comment by jgrumboe** [comment]
@freben could you have a look at this PR, please, if something is missing or it's fine?
Thanks

File: plugins/auth-backend/README.md [hunk 1]
```diff
@@ -34,7 +34,7 @@ Follow this link, [Create new OAuth App](https://github.com/settings/application
 1. Set Application Name to `backstage-dev` or something along those lines.
 1. You can set the Homepage URL to whatever you want to.
 1. The Authorization Callback URL should match the redirect URI set in Backstage.
-   1. Set this to `http://localhost:7007/api/auth/github` for local development.
+   1. Set this to `http://localhost:7007/api/auth/github/handler/frame` for local development.
    1. Set this to `http://{APP_FQDN}:{APP_BACKEND_PORT}/api/auth/github` for non-local deployments.
```
Comment (Rugvip) [inline plugins/auth-backend/README.md:38]: This one should be updated as well

**Review by Rugvip — CHANGES_REQUESTED** [review]
Thanks! Just a minor tweak

File: plugins/auth-backend/README.md (hunk 1 above)
Comment (jgrumboe) [reply plugins/auth-backend/README.md:38]: Thanks, good catch.

Comment (jgrumboe) [reply plugins/auth-backend/README.md]: @Rugvip fixed

**Review by Rugvip — APPROVED** [review]
🎉 , let's :shipit: !

**Comment by suuus** [comment]
Thank you for contributing during Hacktoberfest 🍂! You can claim the Backstage Hacktoberfest Holopin at https://bck.st/hacktoberfest-holopin 🙌🏻

**Pull request merged**